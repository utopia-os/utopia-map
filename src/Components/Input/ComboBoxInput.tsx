interface ComboBoxProps {
  id?: string
  options: string[]
  value: string
  onValueChange: (newValue: string) => void
}

const ComboBoxInput = ({ id, options, value, onValueChange }: ComboBoxProps) => {
  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onValueChange(e.target.value)
  }

  return (
    <select
      id={id}
      className='tw:form-select tw:block tw:w-full tw:py-2 tw:px-4 tw:border tw:border-gray-300 rounded-md tw:shadow-sm tw:text-sm tw:focus:outline-hidden tw:focus:ring-indigo-500 tw:focus:border-indigo-500 tw:sm:text-sm'
      onChange={handleChange}
      value={value} // ← hier controlled statt defaultValue
    >
      {options.map((o) => (
        <option value={o} key={o}>
          {o}
        </option>
      ))}
    </select>
  )
}

export default ComboBoxInput
