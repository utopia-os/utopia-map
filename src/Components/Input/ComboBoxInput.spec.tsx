import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, beforeEach, vi } from 'vitest'

import ComboBoxInput from './ComboBoxInput'

describe('<ComboBoxInput />', () => {
  let wrapper: ReturnType<typeof render>

  const updateFormValue = vi.fn()

  beforeEach(() => {
    vi.clearAllMocks()
    wrapper = render(
      <ComboBoxInput
        value={'option-1'}
        onValueChange={updateFormValue}
        options={['Option 1', 'Option 2']}
      />,
    )
  })

  it('renders properly', () => {
    expect(wrapper.container.firstChild).toMatchSnapshot()
  })

  describe('handleChange', () => {
    it('calls updateFormValue with new value', () => {
      fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Option 2' } })
      expect(updateFormValue).toBeCalledWith('Option 2')
    })
  })
})
