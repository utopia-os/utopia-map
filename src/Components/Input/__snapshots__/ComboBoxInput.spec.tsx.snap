// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ComboBoxInput /> > renders properly 1`] = `
<select
  class="tw-form-select tw-block tw-w-full tw-py-2 tw-px-4 tw-border tw-border-gray-300 rounded-md tw-shadow-sm tw-text-sm focus:tw-outline-none focus:tw-ring-indigo-500 focus:tw-border-indigo-500 sm:tw-text-sm"
>
  <option
    value="Option 1"
  >
    Option 1
  </option>
  <option
    value="Option 2"
  >
    Option 2
  </option>
</select>
`;
